### PR TITLE
Add part command

### PR DESCRIPTION
**Pull request by hbnrmx**
adds part of #5595

### CODE REVIEW

File: osu.Game/Online/Chat/ChannelManager.cs [hunk 1]
```diff
@@ -213,6 +213,11 @@ public void PostCommand(string text, Channel target = null)
                     PostMessage(content, true);
                     break;
 
+                case "p":
```
Comment (peppy) [inline osu.Game/Online/Chat/ChannelManager.cs]: i don't think we should support `/p`.

File: osu.Game/Online/Chat/ChannelManager.cs [hunk 2]
```diff
@@ -213,6 +213,11 @@ public void PostCommand(string text, Channel target = null)
...
+                case "part":
+                    LeaveChannel(target);
+                    break;
+
                 case "help":
                     target.AddNewMessages(new InfoMessage("Supported commands: /help, /me [action]"));
```
Comment (peppy) [inline osu.Game/Online/Chat/ChannelManager.cs]: missing help

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by smoogipoo** [comment]
This is opening up the channel selection overlay when a channel is parted.

**Comment by peppy** [comment]
It probably should if you part the last channel you were joined, but no other case.

**Comment by TPGPL** [comment]
probably should implement the shortened version of the command as well - `/p`

**Comment by jorolf** [comment]
> 
> 
> probably should implement the shortened version of the command as well - `/p`

https://github.com/ppy/osu/pull/5601#discussion_r310405265

**Comment by TPGPL** [comment]
o nevermind then

**Comment by hbnrmx** [comment]
> This is opening up the channel selection overlay when a channel is parted.

Took me a while to figure out why this happens for `/part` and not closing by button. 

(See here: https://github.com/ppy/osu/blob/907cf18960acd6bf460c057dbd60e95f103cbda4/osu.Game/Overlays/ChatOverlay.cs#L232)

Turns out calling `LeaveChannel()` in `ChannelManager` isn't enough. 
Instead I think we want to invoke `ChannelTabControl`'s `OnRequestClose` since all the necessary tab closing logic is handled there.

This feels kinda backwards / cyclic though. Any idea how this should be handled?

**Comment by peppy** [comment]
`ChannelTabControl` should handle leaving a channel only. the request close should just fire the leave action, so everything comes in through a single path.

**Comment by hbnrmx** [comment]
> 
> 
> `ChannelTabControl` should handle leaving a channel only. the request close should just fire the leave action, so everything comes in through a single path.

I can read your comment in two different ways, so just to confirm:

You're saying I should move the tab selection logic 
(https://github.com/ppy/osu/blob/907cf18960acd6bf460c057dbd60e95f103cbda4/osu.Game/Overlays/Chat/Tabs/ChannelTabControl.cs#L108
)
from `ChannelTabControl` to `ChannelManager` ?

**Comment by peppy** [comment]
`ChannelTabControl` should probably be handling operations based on `JoinedChannels` bindable. it's a larger change to make to fix things up (probably a prereq for this PR to happen).

**Pull request closed without merging**